### DISCUSSION TITLE
Moonlight / Sunshine Remote Co-opt  game support

### DISCUSSION THREAD

**Discussion by Silas888** [LizardByte/community#120] [Sunshine] [answered]
I was curious of Moonlight or Sunshine, supports remote co-op, game support, such as "It Takes Two?" Currently I have it set up where Parsec streams my host so my remote guest can play however, I'm curious if moonlight can do the same thing.

**Comment by ReenigneArcher** [comment, marked as answer]
Yes Sunshine supports it. Please see the documentation for "channels" -> https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/advanced_usage.html#channels

**Reply by Silas888** [reply]
Thank you for your reply but I'm still a little confused. I see that I would have to change my channel to equal 2 but I'm not sure how to set up the remote side. Would the remote player need to have their own copy of moonlight? If so, am I correct on assuming they would connect to me via Zero Tier? Would they simply be joining my game in progress from Moonlight or is there another process entirely? Is there any more documentation on this process?

**Reply by ReenigneArcher** [reply]
Every client that connects to Sunshine requires Moonlight.

There's various ways to connect to Sunshine. Zero-Tier is a popular option for sure, but I can't give any guidance on it as I've never used it. Some routers have built in VPN tunnels and you can use Open-VPN for example. Sunshine can also be exposed directly to the internet, although I would really advise against that.

One person would need to start the session, and the other to join the already started session.

**Reply by Silas888** [reply]
Perfect. Thank you. That was the part I was missing. I look forward to trying it.

**Reply by Hukuma1** [reply]
Exposing it to the Internet would be "okay" if you trust the other user. Their instance of Moonlight would still have to be authenticated on your end with a PIN. The slight issue that could come up is some privacy. Since we're simply streaming our entire desktop over to the other user, it can be a bit weird. I see there's workarounds on Discord, but is it possible to block the "Desktop" app entirely from Sunshine allowing? Or at least a setting in the future for X client(s)/users to not be allowed to Desktop app from Moonlight? Would really love to just lock the player for specific app and not let them have desktop.

And as I typed this I realized once you've authenticated someone via PIN, you can't see/revoke it on the host side. Oops. I guess some very basic user/clients display needs to be implemented for this to go into better co-play remote playing scenarios.

**Reply by Silas888** [reply]
I currently use Parsec. What I like about Parsec is that I'm able to add or revoke users at any time. Moreover, I'm allowed to isolate the game or games I wish to share rather than my desktop. That way, if I'm playing another game that is not shared, the remote user will not see anything.
